src/slurm_plugin/clustermgtd.py

```diff
@@ -603,84 +536,42 @@ def _get_ec2_instances(self):
             raise ClusterManager.EC2InstancesInfoUnavailable
 
     @log_exception(log, "performing health check action", catch_exception=Exception, raise_on_error=False)
-    def _perform_health_check_actions(self, cluster_instances, ip_to_slurm_node_map):
+    def _perform_health_check_actions(self, partitions):
         """Run health check actions."""
+        instance_id_to_active_node_map = ClusterManager.get_instance_id_to_active_node_map(partitions)
         log.info("Performing instance health check actions")
-        id_to_instance_map = {instance.id: instance for instance in cluster_instances}
         # Get health states for instances that might be considered unhealthy
         unhealthy_instance_status = self._instance_manager.get_unhealthy_cluster_instance_status(
-            list(id_to_instance_map.keys())
+            list(instance_id_to_active_node_map.keys())
         )
         log.debug("Cluster instances that might be considered unhealthy: %s", unhealthy_instance_status)
         if unhealthy_instance_status:
             # Perform EC2 health check actions
             if not self._config.disable_ec2_health_check:
                 self._handle_health_check(
```

Comment:
_handle_health_check  is counting towards protected mode. In the unlikely event that a node has both a ec2_health faliure and a scheduled_event we risk to count this twice. Btw see my comments about extracting the protected mode logic from here which might address this problem as well.

Reply:
Done